R/estimate_delay.R

```diff
@@ -48,13 +48,29 @@ estimate_delay <- function(
     reporting_triangle,
     max_delay = ncol(reporting_triangle) - 1,
     n = nrow(reporting_triangle)) {
+  if (max_delay > ncol(reporting_triangle) - 1) {
+    message(sprintf(
+      "The maximum delay must be less than the number of columns in the reporting triangle. The maximum delay will be set to %d.", # nolint
+      ncol(reporting_triangle) - 1
+    ))
+    max_delay <- ncol(reporting_triangle) - 1
+  }
+
+  if (max_delay < ncol(reporting_triangle) - 1 && max_delay > 0) {
+    # filter the reporting triangle to be less than the maximum delay
+    reporting_triangle <- reporting_triangle[, 1:(max_delay + 1)]
+    message(sprintf(
```

Comment:
Why do you sometimes use cli tooling for messages, errors and warnings and sometimes not?